nemo/collections/asr/data/audio_to_text.py

```diff
@@ -928,8 +969,19 @@ def get_manifest_sample(self, sample_id):
     def __iter__(self):
         return self._dataset.__iter__()
 
+    def _compute_len(self):
+        if self.shard_manifests and torch.distributed.is_available() and torch.distributed.is_initialized():
+            my_len = torch.tensor(len(self.manifest_processor.collection), dtype=torch.int32).cuda()
+            torch.distributed.all_reduce(my_len)
```

Comment:
Don't you need the reduce op to be max or something for all reduce ?

Reply:
It is sum by default

Reply:
Exactly - sum across all nodes is the value * num nodes. You want max so that all nodes get the same value.

Reply:
Or wait, I suppose sum is correct across each node cause now all nodes have subset of manifest. Ok makes sense then